plugins/sitemesh/src/main/java/org/apache/struts2/sitemesh/VelocityDecoratorServlet.java

```diff
@@ -78,17 +88,18 @@ public void init(ServletConfig config) throws ServletException {
         velocityManager.init(config.getServletContext());
 
         // do whatever we have to do to init Velocity
-        setVelocityEngine(velocityManager.getVelocityEngine());
-        toolboxManager = velocityManager.getToolboxManager();
+        getVelocityView().setVelocityEngine(velocityManager.getVelocityEngine());
+        //getVelocityView().getToolboxFactory(). toolboxManager = velocityManager.getToolboxManager();

```

Comment:
WARN: I didn't find a way to translate this line to the new API, but the ALL test are passing